src/smart-escrow/SmartEscrow.sol

```diff
@@ -0,0 +1,200 @@
+
+    /// @notice The error is thrown when the contract is not terminated, when it should be.
+    error ContractIsNotTerminated();
+
+    /// @notice Set initial parameters.
+    /// @param _beneficiaryOwner Address which can update the beneficiary address.
+    /// @param _beneficiary Address which receives tokens that have vested.
+    /// @param _escrowOwner Address which can terminate the contract.
+    /// @param _start Timestamp of the start of vesting period (or the cliff, if there is one).
+    /// @param _end Timestamp of the end of the vesting period.
+    /// @param _vestingPeriodSeconds Period of time between each vesting event in seconds.
+    /// @param _initialTokens Number of OP tokens which vest at start time.
+    /// @param _vestingEventTokens Number of OP tokens which vest upon each vesting event.
+    constructor(
+        address _beneficiaryOwner,
+        address _beneficiary,
+        address _escrowOwner,
+        uint256 _start,
+        uint256 _end,
+        uint256 _vestingPeriodSeconds,
+        uint256 _initialTokens,
+        uint256 _vestingEventTokens
+    ) {
+        if (_beneficiaryOwner == address(0) || _beneficiary == address(0) || _escrowOwner == address(0)) {
+            revert AddressIsZeroAddress();
+        }
+        if (_start == 0) revert StartTimeIsZero();
+        if (_start > _end) revert StartTimeAfterEndTime(_start, _end);
+        if (_vestingPeriodSeconds == 0) revert VestingPeriodIsZeroSeconds();
+
+        beneficiary = _beneficiary;
+        beneficiaryOwner = _beneficiaryOwner;
+        start = _start;
+        end = _end;
+        vestingPeriod = _vestingPeriodSeconds;
+        initialTokens = _initialTokens;
+        vestingEventTokens = _vestingEventTokens;
+
+        _transferOwnership(_escrowOwner);
+    }
+
+    /// @notice Allow escrow owner (2-of-2 multisig with beneficiary and benefactor
+    ///         owners as signers) to terminate the contract.
+    /// @param _returnAddress Address to send remaining contract holdings to after termination
+    /// @notice Emits a {ContractTerminated} event.
+    function terminate(address _returnAddress) external onlyOwner {
+        contractTerminated = true;
+        withdrawUnvestedTokens(_returnAddress);
+        emit ContractTerminated();
+    }
+
+    /// @notice Allow contract owner to update beneficiary owner address.
+    /// @param _newBeneficiaryOwner Address to send remaining contract holdings to after termination
+    /// @notice Emits a {BeneficiaryOwnerUpdated} event.
+    function updateBeneficiaryOwner(address _newBeneficiaryOwner) external onlyOwner {
+        if (_newBeneficiaryOwner == address(0)) revert AddressIsZeroAddress();
+        if (beneficiaryOwner != _newBeneficiaryOwner) {
+            address oldBeneficiaryOwner = beneficiaryOwner;
+            beneficiaryOwner = _newBeneficiaryOwner;
+            emit BeneficiaryOwnerUpdated(oldBeneficiaryOwner, _newBeneficiaryOwner);
+        }
```

Comment:
Since the happy path will more often than not be hit here (the new beneficiary owner is different than the old) we can save a `SLOAD` by moving the `oldBeneficiaryOwner` declaration above the `if` statement.

```suggestion
        address oldBeneficiaryOwner = beneficiaryOwner;
        if (oldBeneficiaryOwner != _newBeneficiaryOwner) {
            beneficiaryOwner = _newBeneficiaryOwner;
            emit BeneficiaryOwnerUpdated(oldBeneficiaryOwner, _newBeneficiaryOwner);
        }
````